// ng2-translate/src/translate.directive.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-translate/src/translate.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
export class Wrapper_TranslateDirective {
  context:import0.TranslateDirective;
  changed:boolean;
  /*private*/ _translate:any;
  /*private*/ _translateParams:any;
  constructor(p0:any,p1:any) {
    this.changed = false;
    this._translate = import1.UNINITIALIZED;
    this._translateParams = import1.UNINITIALIZED;
    this.context = new import0.TranslateDirective(p0,p1);
  }
  check_translate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._translate,currValue))) {
      this.changed = true;
      this.context.translate = currValue;
      this._translate = currValue;
    }
  }
  check_translateParams(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._translateParams,currValue))) {
      this.changed = true;
      this.context.translateParams = currValue;
      this._translateParams = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    return changed;
  }
}
